worker-message-prioritization-distribution/src/test/java/com/github/workerframework/workermessageprioritization/redistribution/consumption/StagingQueueWeightSettingsTest.java

```diff
@@ -0,0 +1,213 @@
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class StagingQueueWeightSettingsTest {
+
+    @Test
+    public void getQueueWeightTest() {
+
+        final Queue targetQueue = getQueue("tq", 1000);
+
+        final Queue q1 = getQueue("bulk-indexer-in»/clynch/enrichment-workflow", 1000);
+        final Queue q2 = getQueue("bulk-indexer-in»/rory3/enrichment-workflow", 1000);
+        final Queue q3 = getQueue("dataprocessing-classification-in»/clynch/update-entities-workflow", 1000);
+        final Queue q4 = getQueue("dataprocessing-classification-in»/a77777/update-entities-workflow", 1000);
+        final Queue q5 = getQueue("dataprocessing-classification-in»/rory3/update-entities-workflow", 1000);
+        final Queue q6 = getQueue("bulk-indexer-in»/clynch/a77777", 1000);
+        final Queue q7 = getQueue("dataprocessing-langdetect-in»/mahesh/ingestion-workflow", 1000);
+        final Queue q8 = getQueue("bulk-indexer-in»/jmcc02/repository-initialization-workflow", 1000);
+
+        final Set<Queue> stagingQueues = new HashSet<>(Arrays.asList(q1, q2, q3, q4, q5, q6, q7, q8));
+        final DistributorWorkItem distributorWorkItem = mock(DistributorWorkItem.class);
+        when(distributorWorkItem.getStagingQueues()).thenReturn(stagingQueues);
+        when(distributorWorkItem.getTargetQueue()).thenReturn(targetQueue);
+
+        // Mock environment variables to set weights using regex pattern followed by weight.
+        final Map<String, String> envVariables = new HashMap<>();
+
+        envVariables.put("CAF_ADJUST_QUEUE_WEIGHT", "enrichment\\-workflow$,10");
+        envVariables.put("CAF_ADJUST_QUEUE_WEIGHT_1", "clynch,0");
+        envVariables.put("CAF_ADJUST_QUEUE_WEIGHT_3", "a77777,3");
+        envVariables.put("CAF_ADJUST_QUEUE_WEIGHT_4", "dataprocessing\\-langdetect\\-in,7");
+        envVariables.put("CAF_ADJUST_QUEUE_WEIGHT_5", "repository\\-initialization\\-workflow$,0.5");
+
+        try(MockedStatic<EnvVariableCollector> envVariableCollectorMock = Mockito.mockStatic(EnvVariableCollector.class)) {
+
+            envVariableCollectorMock.when(EnvVariableCollector::getEnvVariables).thenReturn(envVariables);
+            envVariableCollectorMock.when(() -> EnvVariableCollector.getQueueWeightEnvVariables(anyMap())).thenReturn(envVariables);
+
+            final StagingQueueWeightSettingsProvider stagingQueueWeightSettingsProvider =
+                    new StagingQueueWeightSettingsProvider();
+
+            final List<String> stagingQueueNames =
+                    distributorWorkItem.getStagingQueues().stream().map(Queue::getName).collect(toList());
+
+            final Map<String, Double> stagingQueueWeightMap =
+                    stagingQueueWeightSettingsProvider.getStagingQueueWeights(stagingQueueNames);
+
+            assertEquals("Weight of queue should be set by environment variable.",
```

Comment:
When asserting multiple things the ErrorCollector is a good approach, something I should use more myself.
See the examples in TargetQueueLengthRounderTest.

When you end up with a lot of asserts it can be a code smell that we're testing more than one scenario at a time. So this a situation where we'd want to have multiple unit tests for the different outcomes, but I see how the amount of "arranging" code would tempt you to use a single test. One technique to reduce the arranging code is to introduce some helper methods so you don't have to duplicate code in lots of unit tests, a bit like what you did with the getQueue helper methods.

I'm fine to leave this as is.